src/pipeline/rackhd/docker/Publish.groovy

```diff
@@ -0,0 +1,50 @@
+package pipeline.rackhd.docker
+
+/*
+publish rackhd
+:library_dir: the directory of on-build-config
+:docker_tar_path: where the docker image tar lies
+:dockerhub_user: the rackhd user of dockerhub
+:dockerhub_pass: the rackhd user password of dockerhub
+*/
+def publish(String library_dir, String docker_tar_path, String dockerhub_user, String dockerhub_pass) {
```

Comment:
is it better to pass jenkins creds id here instead of user/password?

Reply:
waiting to discuss.

Reply:
I'm fine with @Qiang's implementation :-)
usability and flexibility , either way .